Automattic/jetpack
# Jetpack Sync: Add endpoint for resetting Sync locks

Adds a new Sync endpoint, `DELETE /jetpack/v4/sync/locks`, that enables resetting all Sync locks.
Useful for debugging purposes.

#### Changes proposed in this Pull Request:

* `Automattic\Jetpack\Sync\Actions`: Adds a new public static method, `reset_locks`, responsible for resetting all Sync related locks
* `Automattic\Jetpack\Sync\REST_Endpoints`: Adds a new endpoint,`DELETE /jetpack/v4/sync/locks`, for resetting all Sync locks.

#### Other information:


#### Jetpack product discussion


p9dueE-6cR-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






- Check the added unit tests for test cases: 1) Calling the endpoint w/without permissions
- Make sure we've covered all the Sync locks
- Try setting the Sync locks to some extreme values and verify they are reset when calling the endpoint*

_* One way of testing this is to use Jetpack Debug's REST API Tester module so in case you plan to test with a JN site, make sure to include the Jetpack Debug plugin_
